Add fallbackToDownoadAction

leshikus opened the pull request:
Hello Dawid,
This is PR for https://github.com/dawidd6/action-download-artifact/issues/173
The only notable changes are in `main.js`, a behavior changes if workflow_conclusion is 'in_progress' or empty
It has some limitations, e.g. no control over unpacking, search_artifacts is ignored when checking the current build
This is my first attempt in modern js, feel free outline if there are any style errors or other errors

dawidd6 commented on main.js:
```diff
@@ -87,6 +104,19 @@ async function main() {
             core.info(`==> Run number: ${runNumber}`)
         }
 
+        if (!workflowConclusion || workflowConclusion == 'in_progress') {
```
Shouldn't this be done only if matching workflow run is not found? So it can actually be a fallback, rather than just a conditional behavior.

leshikus replied:
Hi Dawid, 
this comes from the concept that the most fresh artifact is returned. Also this fits exactly to my use case when I have to use the artifact containing the most recently updated cache. So I would vote for keeping things like this or using search_artifacts parameter to switch this behavior on.

dawidd6 replied:
Ye, I would like this behavior to be opt-in with some dedicated input.

leshikus replied:
It seems my case corresponds to check_artifacts == true

leshikus replied:
The description says,
```
    # check the workflow run to whether it has an artifact
    # then will get the last available artifact from the previous workflow
    # default false, just try to download from the last one```

dawidd6 commented on main.js:
```diff
@@ -1,10 +1,27 @@
 const pathname = require('path')
 const fs = require('fs')
 
+async function fallbackToDownoadAction(name, path) {
```
```suggestion
async function fallbackToDownloadAction(name, path) {
```

leshikus replied:
will fix

leshikus commented:
Dawid, if I misread the meaning of check_artifacts just let me know; i will create another flag, or another flag value for that

dawidd6 commented:
Another input would be much nicer.

leshikus commented:
Hey Dawid, I've simplified PR to be a "fallback only". I've also removed one of three tests. check_artifacts is no longer altered. Could you please take a look? @dawidd6

dawidd6 commented:
Thanks, additionally made some cosmetical changes only.

dawidd6 commented:
This change broke CI on master: https://github.com/dawidd6/action-download-artifact/actions/runs/3170303976

Reverting...

dawidd6 commented:
...and it remains broken after reverting, interesting.

dawidd6 commented:
Actually something else broke, reverted second time, so it stays merged :smile:

leshikus commented:
@dawidd6 thank you for merging and for the great and very useful action
